### ISSUE TITLE
Make LEDC frequency more accurate (IDFGH-6037)

### ISSUE THREAD

**Issue by IhorNehrutsa** [espressif/esp-idf#7722]
Function ledc_set_timer_div() calculates the LEDC timer divisor 
https://github.com/espressif/esp-idf/blob/b86fe0c66ca17d5907ae26ecdffc9d4eb79cec07/components/driver/ledc.c#L261-L314

It uses an integer division with truncating the fractional portion of target divisor. See lines:
https://github.com/espressif/esp-idf/blob/b86fe0c66ca17d5907ae26ecdffc9d4eb79cec07/components/driver/ledc.c#L275
https://github.com/espressif/esp-idf/blob/b86fe0c66ca17d5907ae26ecdffc9d4eb79cec07/components/driver/ledc.c#L280
https://github.com/espressif/esp-idf/blob/b86fe0c66ca17d5907ae26ecdffc9d4eb79cec07/components/driver/ledc.c#L284
https://github.com/espressif/esp-idf/blob/b86fe0c66ca17d5907ae26ecdffc9d4eb79cec07/components/driver/ledc.c#L293

It is possible to get more accurate LEDC frequency by rounding the fractional part of the divisor **div_param**.
It can be made by 
```
 div_param = ( (uint64_t) XXXXXXXX + (freq_hz * precision) / 2 ) / (freq_hz * precision); 
```
like
```
div_param = ( ((uint64_t) s_ledc_slow_clk_8M << 8) + (freq_hz * precision) / 2) / (freq_hz * precision);
```

For example, to round division by 10 use
```
int rounded = (XXX + 10/2 ) / 10
```

**Comment by o-marshmallow** [comment]
Thanks for your feedback, I am currently handling this issue.

**Comment by IhorNehrutsa** [comment]
```
#define ROUNDED_DIVISION(dividend, divisor) (((dividend) + (divisor) / 2) / (divisor))
```

how to use:
```
div_param = ROUNDED_DIVISION((uint64_t) s_ledc_slow_clk_8M << 8, freq_hz * precision);
```

**Comment by IhorNehrutsa** [comment]
Hello.
Is it progress in this issue?
Is it possible to fix be applied from the ESP-IDF v4.0 release up to and including the latest versions?

Thanks for your work.

**Comment by o-marshmallow** [comment]
Hello @IhorNehrutsa ,
Sorry for the delay, I am currently working on this, I have pushed a fix that should be merged soon, I will then check whether this can be backported up to v4.0
